src/orange/orangeinp/Solid.cc

```diff
@@ -0,0 +1,169 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/orangeinp/Solid.cc
+//---------------------------------------------------------------------------//
+#include "Solid.hh"
+
+#include "corecel/io/JsonPimpl.hh"
+#include "corecel/math/Algorithms.hh"
+
+#include "ConvexSurfaceBuilder.hh"
+#include "CsgTreeUtils.hh"
+
+#include "detail/BoundingZone.hh"
+#include "detail/BuildConvexRegion.hh"
+#include "detail/CsgUnitBuilder.hh"
+#include "detail/VolumeBuilder.hh"
+
+#if CELERITAS_USE_JSON
+#    include "ObjectIO.json.hh"
+#endif
+
+namespace celeritas
+{
+namespace orangeinp
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct from a starting angle and interior angle.
+ */
+SolidEnclosedAngle::SolidEnclosedAngle(Turn start, Turn interior)
+    : start_{start}, interior_{interior}
+{
+    CELER_VALIDATE(interior_ > zero_quantity() && interior_ <= Turn{1},
+                   << "invalid interior angle " << interior.value()
+                   << " [turns]: must be in (0, 1]");
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Construct a wedge shape to intersect (inside) or subtract (outside).
+ */
+auto SolidEnclosedAngle::make_wedge() const -> SenseWedge
+{
+    CELER_EXPECT(*this);
+    // Get the start value between [0, 1)
+    real_type start = eumod(start_.value(), real_type{1});
+    real_type interior = interior_.value();
+    Sense sense = Sense::inside;
+    if (interior > real_type{0.5})
+    {
+        // Subtract the complement of the wedge
+        sense = Sense::outside;
+        start = eumod(start + interior, real_type{1});
+        interior = 1 - interior;
+    }
+
+    return {sense, InfWedge{Turn{start}, Turn{interior}}};
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Construct a volume from this shape.
+ */
+NodeId SolidBase::build(VolumeBuilder& vb) const
+{
+    std::vector<NodeId> nodes;
+
+    // Build the outside-of-the-shell node
+    nodes.push_back(build_convex_region(
+        vb, std::string{this->label()}, "interior", this->interior()));
+
+    if (auto* exclu = this->excluded())
+    {
+        // Construct the excluded region by building a convex solid, then
+        // negating it
+        NodeId smaller = build_convex_region(
+            vb, std::string{this->label()}, "excluded", *exclu);
+        nodes.push_back(vb.insert_region({}, Negated{smaller}));
+    }
+
+    if (auto const& sea = this->enclosed_angle())
```

Comment:
I don't understand what is going on here. Won't this assignment operation always evaluate to true?

Reply:
This gets a reference to the enclosed angle, which is false if there's no azimuthal truncation.

Reply:
Could you add a comment for that?